### DISCUSSION TITLE
Conv. auto-encoders using mirroring

### DISCUSSION THREAD

**Discussion by patnicolas** [deepjavalibrary/djl#1805] [Deep Learning]
I am working on automating the generation of convolutional auto-encoders/GANs (Java/Scala). The plan is to generate the corresponding de-convolutional (Transposed) automatically from existing block builders.
I am new to DJL but I have not seen such capability in 0.17.0? If not, is there a plan add such functionality in a future release?
I enjoy this 'lego' design to build neural nets..

**Comment by zachgk** [comment]
We do have a `Deconvolution` class that you can use. The use of GAN may be a bit tricky with `Trainer`s, but you can simply avoid the `Trainer` and use the underlying pieces (`GradientCollector`) directly.

Is there a particular area that you are having problems?

**Reply by patrick-nicolas** [reply]
Very sorry for this late reply. I was able to resolve the issue looking at the Gradient collector code. Great library!
